src/cljs/witan/ui/components/create_rts.cljs

```diff
@@ -0,0 +1,232 @@
+(ns witan.ui.components.create-rts
+  (:require [reagent.core :as r]
+            [witan.ui.data :as data]
+            [witan.ui.components.shared :as shared]
+            [witan.ui.components.icons :as icons]
+            [witan.ui.strings :refer [get-string]]
+            [witan.ui.controller :as controller])
+  (:require-macros [cljs-log.core :as log]))
+
+(defn assess-form-validity
+  "A very crude way to validate we have enough information to make an
+   RTS/RFD"
+  [{:keys [recipients schema destinations]}]
+  (let [recipients-result (when-not (pos? (count recipients)) :recipients)
+        schema-result (when-not schema :schema)
+        destinations-result (when-not (pos? (count destinations)) :destinations)]
+    (set (keep identity [recipients-result
+                         schema-result
+                         destinations-result]))))
+
+(defn user-search-area
+  [recipients]
+  (let [results (:user/group-search-results (data/get-app-state :app/user))
+        ids (set (map :kixi.group/id @recipients))]
+    [:div.user-list
+     (for [{:keys [kixi.group/name kixi.group/id] :as group} (sort-by :kixi.group/type results)
+           :when (not (contains? ids id))]
+       [:div.user-list-item {:key name}
+        (shared/button {:icon icons/plus
+                        :id id
+                        :prevent? true}
+                       #(swap! recipients conj group))
+        (shared/inline-group group)])]))
+
+(defn schema-search-area
+  [selected-schema]
+  (let [results (:schema/search-results (data/get-app-state :app/datastore))]
+    [:table.schema-list.pure-table
+     [:thead
+      [:tr
+       [:th]
+       [:th "Name"]
+       [:th "Author"]]]
+     [:tbody
+      (doall
+       (for [{:keys [schema/name schema/id schema/author] :as schema} results
+             :let [author-name (:kixi.group/name author)]
+             :when (not= schema @selected-schema)]
+         [:tr.schema-list-item {:key name}
+          [:td
+           (shared/button {:icon icons/tick
+                           :id id
+                           :prevent? true}
+                          #(reset! selected-schema schema))]
+          [:td
+           [:span name]]
+          [:td author-name]]))]]))
+
+(defn view
+  [this]
+  (let [show-recipient-breakout?   (r/atom false)
+        show-schema-breakout?      (r/atom false)
+        show-destination-breakout? (r/atom false)
+        recipients                 (r/atom [])
+        destinations               (r/atom [])
+        schema                     (r/atom nil)
+        invalid-fields             (r/atom #{})
```

Comment:
This view is essentially a big form. For a few reasons it's better to have these atoms here (or a single atom hashmap, which would be an alternative) but mainly because the transient values in the form are disinteresting to the 'app-state' (big atom) - they don't actually contribute anything.
It's also much easier, for example, for another programmer to grok what's happening here versus having to pick through the app-state.